source/jormungandr/jormungandr/ptref.py

```diff
@@ -0,0 +1,53 @@
+#  Copyright (c) 2001-2014, Canal TP and/or its affiliates. All rights reserved.
+#
+# This file is part of Navitia,
+#     the software to build cool stuff with public transport.
+#
+# Hope you'll enjoy and contribute to this project,
+#     powered by Canal TP (www.canaltp.fr).
+# Help us simplify mobility and open public transport:
+#     a non ending quest to the responsive locomotion way of traveling!
+#
+# LICENCE: This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+
+from __future__ import absolute_import, print_function, unicode_literals, division
+from navitiacommon import request_pb2, type_pb2
+
+
+class PtRef(object):
+
+    def __init__(self, instance):
+        self.instance = instance
+
+    def get_stop_point(self, code_key, code_value):
+        req = request_pb2.Request()
+        req.requested_api = type_pb2.PTREFERENTIAL
+        req.ptref.requested_type = type_pb2.STOP_POINT
+        req.ptref.count = 1
+        req.ptref.start_page = 0
+        req.ptref.depth = 0
+        req.ptref.filter = "stop_point.has_code({code_key}, {code_value})".\
+            format(code_key=code_key, code_value=code_value)
+
+        result = self.instance.send_and_receive(req)
+        if len(result.stop_points) > 0:
```

Comment:
as mentioned: we should check `== 1`, and add a filter on lines to avoid as doubloons as much as possible